validation_ingress.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"fmt"
+	extv1beta1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"regexp"
+	"strings"
+)
+
+var ingressHostRegExp = regexp.MustCompile(`^([a-zA-Z0-9_][a-zA-Z0-9_-]{0,62})(\.[a-zA-Z0-9_][a-zA-Z0-9_-]{0,62})*[._]?$`)
+var ingressPathRegExp = regexp.MustCompile(`^[A-zA-Z0-9_/.\-]*$`)
+
+type TlsDefinition struct {
+	host             string
+	secretName       string
+	ingressName      string
+	ingressNamespace string
+}
+
+type PathDefinition struct {
+	host             string
+	path             string
+	serviceName      string
+	servicePort      string
+	ingressName      string
+	ingressNamespace string
+}
+
+func (pathDefinition *PathDefinition) toUri() string {
+	return pathDefinition.host + pathDefinition.path
+}
+
+func ValidateIngress(validation *objectValidation, ingress *extv1beta1.Ingress, config *config) error {
+
+	targetDesc := fmt.Sprintf("Ingress %s.%s: ", ingress.Name, ingress.Namespace)
+
+	remoteIngresses, err := ingressClient().List(metav1.ListOptions{})
```

Comment:
Does this really refer to "remote" ingresses? That name gives me the impression that it refers to ingresses in a different cluster than one where resources are being validated, which is kind of confusing.

Reply:
renamed to new vs. existing (https://martinfowler.com/bliki/TwoHardThings.html)